Dockerfile

```diff
@@ -1,25 +1,49 @@
 FROM python:3.9-slim-bullseye as base
-ENV POETRY_PATH=/opt/poetry \
-    POETRY_VERSION=1.1.6
-ENV PATH="$POETRY_PATH/bin:$VENV_PATH/bin:$PATH"
+ENV POETRY_HOME=/etc/poetry \
+    POETRY_VERSION=1.2.1
+ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
 
 FROM base AS build
 
 RUN apt-get update && \
     apt-get install -y -q build-essential \
     curl
 
-RUN curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python
-RUN mv /root/.poetry $POETRY_PATH
+# For pyodbc
+RUN curl -s -o microsoft.asc https://packages.microsoft.com/keys/microsoft.asc \
+    && curl -s -o mssql-release.list https://packages.microsoft.com/config/debian/11/prod.list \
+    && apt-get update -y \
+    && apt-get install -y g++ unixodbc-dev
+
+RUN curl -sSL https://install.python-poetry.org | POETRY_HOME=${POETRY_HOME} python3 -
 RUN poetry config virtualenvs.create false
 RUN poetry config experimental.new-installer false
+RUN echo 'export PATH="$PATH:/opt/mssql-tools18/bin"' >> ~/.bashrc
 
 COPY poetry.lock pyproject.toml ./
 RUN poetry install --no-interaction --no-ansi --no-dev -vvv
 
 
 FROM base as runtime
 
+# For pyodbc
+COPY --from=build microsoft.asc microsoft.asc
+COPY --from=build mssql-release.list mssql-release.list
+
+ENV ACCEPT_EULA=Y
+RUN apt-get update -y && apt-get install -y gnupg2
+RUN apt-key add microsoft.asc && rm microsoft.asc && mv mssql-release.list /etc/apt/sources.list.d/mssql-release.list
+RUN apt-get update -y
```

Comment:
Clean Docker filer. You have already used `apt-get update` above

Reply:
Done